Add missing -dev files

sergio-costas opened the pull request:
Some packages are added to the stage, but not their corresponding development packages. This patch fixes this.

Should solve https://github.com/ubuntu/gnome-contacts/pull/9

sergio-costas commented:
Tested with:

    cheese
    chromium (youtube video)
    darktable
    eog
    epiphany (youtube video)
    evince
    folio (still fails)
    gimp
    gnome-characters
    gnome-mahjongg
    gnome-mines
    gnome-recipes
    gnome-system-monitor
    gnome-text-editor
    kicad
    shotwell
    zoom-client
    mattermost-desktop
    element-desktop

sergio-costas commented:
> It's good, but why we are listing both the library without -dev and with -dev, given that the latter depends on the former? Will Snapcraft only stage stuff explictly present in this list?

You are right. I'll fix it.

aleasto commented on snapcraft.yaml:
```diff
@@ -1430,27 +1436,32 @@ parts:
       - libglu1-mesa-dev
       - libgmp10
+      - libgmp-dev
       - libgnutls28-dev
       - libgnutls30t64
       - libgnutls-dane0t64
       - libgspell-1-dev
-      - libgstreamer-plugins-base1.0-0
-      - libgstreamer-plugins-bad1.0-0
-      - libgstreamer-plugins-good1.0-0
+      - libgstreamer-plugins-base1.0-dev
+      - libgstreamer-plugins-bad1.0-dev
+      - libgstreamer-plugins-good1.0-dev
```
What's the use for these? Was this supposed to be "gstreamer1.0-plugins-good" etc.., which are the actual gstreamer plugins?

The lib* packages only provide libraries used by said plugins. The package description is
> This package contains shared GStreamer libraries from the "good" set. The API is not guaranteed to be stable.

And `apt rdepends` shows that indeed they're only used by the gstreamer1.0-plugins-* packages.

aleasto replied:
> The lib* packages only provide libraries used by said plugins.

> And apt rdepends shows that indeed they're only used by the gstreamer1.0-plugins-* packages.

Mh, no, that's the case for the "good" set but not for the "base" or "bad" set

sergio-costas replied:
Well, this PR was created as a fix for the problem commented at https://github.com/ubuntu/gnome-contacts/pull/9

sergio-costas replied:
@aleasto So... Opinion? Should I merge it?

aleasto replied:
Yes

sergio-costas replied:
Great!

sergio-costas commented:
@nteodosio @3v1n0 I sorted all the -dev packages and removed the "duplicated ones" (this is, those runtime packages which already had the corresponding devel packages), and the result was several soft links missing. That's why I added the required ones again...

```
raster ~/workspace/gnome-sdk (include-missing-devel-files)$ diff root.txt upstream.txt 
145a146
> ./lib/x86_64-linux-gnu/liblzo2.so.2
13248a13250,13251
> ./usr/lib/tmpfiles.d
> ./usr/lib/tmpfiles.d/libselinux1.conf
13904a13908,13911
> ./usr/lib/x86_64-linux-gnu/krb5
> ./usr/lib/x86_64-linux-gnu/krb5/plugins
> ./usr/lib/x86_64-linux-gnu/krb5/plugins/preauth
> ./usr/lib/x86_64-linux-gnu/krb5/plugins/preauth/spake.so
14085a14093,14094
> ./usr/lib/x86_64-linux-gnu/libblkid.so.1
> ./usr/lib/x86_64-linux-gnu/libblkid.so.1.1.0
14093a14103,14104
> ./usr/lib/x86_64-linux-gnu/libbrotlicommon.so.1
> ./usr/lib/x86_64-linux-gnu/libbrotlicommon.so.1.1.0
14095a14107,14108
> ./usr/lib/x86_64-linux-gnu/libbrotlidec.so.1
> ./usr/lib/x86_64-linux-gnu/libbrotlidec.so.1.1.0
14097a14111,14112
> ./usr/lib/x86_64-linux-gnu/libbrotlienc.so.1
> ./usr/lib/x86_64-linux-gnu/libbrotlienc.so.1.1.0
14099a14115,14117
> ./usr/lib/x86_64-linux-gnu/libbz2.so.1
> ./usr/lib/x86_64-linux-gnu/libbz2.so.1.0
> ./usr/lib/x86_64-linux-gnu/libbz2.so.1.0.4
14182a14201,14202
> ./usr/lib/x86_64-linux-gnu/libdbus-1.so.3
> ./usr/lib/x86_64-linux-gnu/libdbus-1.so.3.32.4
14232a14253,14254
> ./usr/lib/x86_64-linux-gnu/libdrm.so.2
> ./usr/lib/x86_64-linux-gnu/libdrm.so.2.4.0
14255a14278,14279
> ./usr/lib/x86_64-linux-gnu/libevdev.so.2
> ./usr/lib/x86_64-linux-gnu/libevdev.so.2.3.0
14265a14290,14291
> ./usr/lib/x86_64-linux-gnu/libexpat.so.1
> ./usr/lib/x86_64-linux-gnu/libexpat.so.1.9.1
14267a14294,14295
> ./usr/lib/x86_64-linux-gnu/libexpatw.so.1
> ./usr/lib/x86_64-linux-gnu/libexpatw.so.1.9.1
14290a14319,14320
> ./usr/lib/x86_64-linux-gnu/libfreetype.so.6
> ./usr/lib/x86_64-linux-gnu/libfreetype.so.6.20.1
14321a14352,14353
> ./usr/lib/x86_64-linux-gnu/libgcrypt.so.20
> ./usr/lib/x86_64-linux-gnu/libgcrypt.so.20.4.3
14438a14471,14472
> ./usr/lib/x86_64-linux-gnu/libgmp.so.10
> ./usr/lib/x86_64-linux-gnu/libgmp.so.10.5.0
14477a14512,14513
> ./usr/lib/x86_64-linux-gnu/libgpg-error.so.0
> ./usr/lib/x86_64-linux-gnu/libgpg-error.so.0.34.0
14808a14845,14846
> ./usr/lib/x86_64-linux-gnu/libidn2.so.0
> ./usr/lib/x86_64-linux-gnu/libidn2.so.0.4.0
14874a14913,14914
> ./usr/lib/x86_64-linux-gnu/libkrb5.so.3
> ./usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
14902a14943,14944
> ./usr/lib/x86_64-linux-gnu/liblzma.so.5
> ./usr/lib/x86_64-linux-gnu/liblzma.so.5.4.5
14937a14980,14981
> ./usr/lib/x86_64-linux-gnu/libmount.so.1
> ./usr/lib/x86_64-linux-gnu/libmount.so.1.1.0
15315a15360,15361
> ./usr/lib/x86_64-linux-gnu/libpcre2-8.so.0
> ./usr/lib/x86_64-linux-gnu/libpcre2-8.so.0.11.2
15479a15526,15527
> ./usr/lib/x86_64-linux-gnu/libseccomp.so.2
> ./usr/lib/x86_64-linux-gnu/libseccomp.so.2.5.5
15484a15533
> ./usr/lib/x86_64-linux-gnu/libselinux.so.1
15488a15538
> ./usr/lib/x86_64-linux-gnu/libsepol.so.2
15533a15584,15585
> ./usr/lib/x86_64-linux-gnu/libsqlite3.so.0
> ./usr/lib/x86_64-linux-gnu/libsqlite3.so.0.8.6
15559a15612,15613
> ./usr/lib/x86_64-linux-gnu/libsystemd.so.0
> ./usr/lib/x86_64-linux-gnu/libsystemd.so.0.38.0
15565a15620,15621
> ./usr/lib/x86_64-linux-gnu/libtasn1.so.6
> ./usr/lib/x86_64-linux-gnu/libtasn1.so.6.6.3
15625a15682,15683
> ./usr/lib/x86_64-linux-gnu/libudev.so.1
> ./usr/lib/x86_64-linux-gnu/libudev.so.1.7.8
16026a16085,16086
> ./usr/lib/x86_64-linux-gnu/libwrap.so.0
> ./usr/lib/x86_64-linux-gnu/libwrap.so.0.7.6
16129a16190,16191
> ./usr/lib/x86_64-linux-gnu/libxkbcommon.so.0
> ./usr/lib/x86_64-linux-gnu/libxkbcommon.so.0.0.0
16167a16230,16231
> ./usr/lib/x86_64-linux-gnu/libyaml-0.so.2
> ./usr/lib/x86_64-linux-gnu/libyaml-0.so.2.0.9
16173a16238,16239
> ./usr/lib/x86_64-linux-gnu/libz.so.1
> ./usr/lib/x86_64-linux-gnu/libz.so.1.3
31850a31917,31918
> ./usr/share/libgcrypt20
> ./usr/share/libgcrypt20/clean-up-unmanaged-libraries
```

nteodosio commented:
So it look like Snapcraft indeed only considers staging what is in the package list. I'd suggest putting a comment with your findings. But good to go anyways for me.

sergio-costas commented:
> So it look like Snapcraft indeed only considers staging what is in the package list. I'd suggest putting a comment with your findings. But good to go anyways for me.

That's a good point, indeed. Give me a second...

sergio-costas commented:
@nteodosio Ok, I found why... those packages are already in Core24. That's why it doesn't install them.

So... maybe we can remove them without problem.
